HW2/P2/parallel_vector.pyx

```diff
@@ -107,9 +113,20 @@ cpdef move_data_medium_grained(np.int32_t[:] counts,
    ##########
    with nogil:
        for r in range(repeat):
-           for idx in range(src.shape[0]):
+           for idx in prange(src.shape[0], num_threads=4):
                if counts[src[idx]] > 0:
-                   counts[dest[idx]] += 1
-                   counts[src[idx]] -= 1
+                   if src[idx]/N == dest[idx]/N:
```

Comment:
src[idX]/N == dest [idx]/N  <=> src[idX] == dest [idx]  (the N for comparison is not necessary)